File: cam_benchmark.py
import sys

SIZE = 50000

A = [i for i in range(SIZE)]
B = [50000 - i for i in range(SIZE)]
C = [0] * SIZE

for i in range(SIZE):
    C[i] = sum(B[:A[i] + 1])

checksum = sum(C)

print(checksum)

if checksum != 41667916675000:
    print("Error: Computed checksum does not match 41667916675000.", file=sys.stderr)
    sys.exit(1)

File: cam_benchmark.c
#include <stdio.h>


static int A[50000];
static int B[50000];
static int C[50000];


int main(void) {
    for (int i = 0; i < 50000; i++){
        A[i] = i;
        B[i] = 50000 - i;
        C[i] = 0;
    }


    for ( int i = 0; i < 50000; i++){
        int sum = 0;
        for ( int j = 0; j< A[i]; j++){
            sum += B[j];
        }
        C[i] = sum;


    }
    long long checksum = 0;
    for (int i =0; i < 50000; i++){
        checksum += C[i];
    }
    printf("%lld\n", checksum);
    if (checksum != 41667916675000){
        printf("checksum does not equal: 41667916675000");
        return 1;
    }
    return 0;
}
